plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/IcebergQueryRunner.java

```diff
@@ -238,6 +245,56 @@ public static void main(String[] args)
         }
     }
 
+    public static final class IcebergAzureQueryRunnerMain
+    {
+        private IcebergAzureQueryRunnerMain() {}
+
+        public static void main(String[] args)
+                throws Exception
+        {
+            String azureContainer = requireNonNull(System.getenv("AZURE_CONTAINER"), "Environment variable AZURE_CONTAINER must be provided");
+            String azureAccount = requireNonNull(System.getenv("AZURE_ACCOUNT"), "Environment variable AZURE_ACCOUNT must be provided");
+            String azureAccessKey = requireNonNull(System.getenv("AZURE_ACCESS_KEY"), "Environment variable AZURE_ACCESS_KEY must be provided");
+
+            String abfsSpecificCoreSiteXmlContent = Resources.toString(Resources.getResource("hdp3.1-core-site.xml.abfs-template"), UTF_8)
+                    .replace("%ABFS_ACCESS_KEY%", azureAccessKey)
+                    .replace("%ABFS_ACCOUNT%", azureAccount);
+
+            FileAttribute<Set<PosixFilePermission>> posixFilePermissions = PosixFilePermissions.asFileAttribute(PosixFilePermissions.fromString("rw-r--r--"));
+            Path hadoopCoreSiteXmlTempFile = java.nio.file.Files.createTempFile("core-site", ".xml", posixFilePermissions);
+            hadoopCoreSiteXmlTempFile.toFile().deleteOnExit();
+            java.nio.file.Files.writeString(hadoopCoreSiteXmlTempFile, abfsSpecificCoreSiteXmlContent);
+
+            @SuppressWarnings("resource")
+            HiveMinioDataLake hiveMinioDataLake = new HiveMinioDataLake(
+                    "test-bucket",
+                    ImmutableMap.of("/etc/hadoop/conf/core-site.xml", hadoopCoreSiteXmlTempFile.normalize().toAbsolutePath().toString()),
+                    HiveHadoop.HIVE3_IMAGE);
+            hiveMinioDataLake.start();
+
+            DistributedQueryRunner queryRunner = IcebergQueryRunner.builder()
+                    .setCoordinatorProperties(Map.of(
+                            "http-server.http.port", "8080"))
+                    .setIcebergProperties(Map.of(
+                            "iceberg.catalog.type", "HIVE_METASTORE",
+                            "hive.metastore.uri", "thrift://" + hiveMinioDataLake.getHiveHadoop().getHiveMetastoreEndpoint(),
+                            "hive.azure.abfs-storage-account", azureAccount,
+                            "hive.azure.abfs-access-key", azureAccessKey))
+                    .setSchemaInitializer(
+                            SchemaInitializer.builder()
+                                    .withSchemaName("tpch")
+                                    .withClonedTpchTables(TpchTable.getTables())
+                                    .withSchemaProperties(Map.of("location", "'abfs://%s@%s.dfs.core.windows.net/test-bucket/'", azureContainer, azureAccount))
+                                    .build())
+                    .build();
+
+            Thread.sleep(10);
+            Logger log = Logger.get(IcebergQueryRunner.class);
```

Comment:
```suggestion
            Logger log = Logger.get(IcebergAzureQueryRunnerMain.class);
```
(I know the existing main methods use `IcebergQueryRunner` though)

Reply:
Addressed the other ones in a second commit.